junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertTimeoutAssertionsTests.kt

```diff
@@ -270,11 +270,11 @@ internal class KotlinAssertTimeoutAssertionsTests {
      * Take a nap for 100 milliseconds.
      */
     private fun nap() {
-        val start = System.currentTimeMillis()
+        val start = System.nanoTime()
         // workaround for imprecise clocks (yes, Windows, I'm talking about you)
         do {
             Thread.sleep(100)
-        } while (System.currentTimeMillis() - start < 100)
+        } while (System.nanoTime() - start < 100_000_000)
```

Comment:
Since this is not related to `throwAsUncheckedException()`, please move these changes to a separate PR with the rationale for the change.

Reply:
Reverted.